__tests__/PermissionsList.test.ts

```diff
@@ -0,0 +1,27 @@
+import type { PermissionsList, Permission } from "./types";
+
+describe("PermissionsList class", () => {
+  it("add new element to PermissionsList", () => {
+    let permission: Permission;
+    let list: PermissionsList;
+    // @ts-expect-error
+    list.addPermission(permission);
+
+    // @ts-expect-error
+    expect(list.permissions[0]).toBe(permission);
+  });
+
+  it("removes permission from PermissionsList", () => {
+    let permission1: Permission;
+    let permission2: Permission;
+    let list: PermissionsList;
+    // @ts-expect-error
+    list.addPermission(permission1);
+    // @ts-expect-error
+    list.addPermission(permission2);
+    // @ts-expect-error
+    list.removePermission(permission2);
+    // @ts-expect-error
+    expect(list.permissions[0]).toBe(permission1);
```

Comment:
Checking an exact position of the array doesn't seem to be a good idea. Referring to [jest docs](https://jestjs.io/docs/expect#tocontainitem) it seems that `expect(list.permissions).toContain(permission1)` is the right approach here.